packages/react-import-remote/src/ImportRemote.tsx

```diff
@@ -50,24 +62,45 @@ export default class ImportRemote extends React.PureComponent<Props, never> {
   }
 
   render() {
-    const {source, preconnect} = this.props;
+    const {loaded} = this.state;
+    const {source, preconnect, defer} = this.props;
+
+    const intersectionObserver =
+      !loaded && defer === DeferTiming.InViewport ? (
+        <IntersectionObserver
+          threshold={0}
+          unsupportedBehavior={UnsupportedBehavior.TreatAsIntersecting}
+          onIntersecting={this.loadRemote}
+        />
+      ) : null;
 
     if (preconnect) {
       const url = new URL(source);
-      return <Preconnect source={url.origin} />;
+      return (
+        <>
+          <Preconnect source={url.origin} />
+          {intersectionObserver}
+        </>
+      );
     }
 
-    return null;
+    return intersectionObserver;
   }
 
-  async loadRemote() {
-    const {source, nonce = '', getImport, onError, onImported} = this.props;
+  private loadRemote = () => {
+    return new Promise(resolve => {
+      this.setState({loaded: false}, async () => {
+        const {source, nonce = '', getImport, onError, onImported} = this.props;
 
-    try {
-      const imported = await load(source, getImport, nonce);
-      onImported(imported);
-    } catch (error) {
-      onError(error);
-    }
-  }
+        try {
+          const imported = await load(source, getImport, nonce);
+          onImported(imported);
```

Comment:
Is `onImported` going to be called every time `<ImportRemote defer={DeferTiming.InViewport}>` scrolls into view?    It feels like the kind of thing that should only be called once.

Reply:
No, we had logic to remove this once the module had loaded. To make it more bulletproof, I've added a `loading` piece to state that also hides it during the process of loading the module.